applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -1025,6 +1046,170 @@ static void send_enqueued_report(struct subscriber *sub)
 	}
 }
 
+static void hogp_out_rep_write_cb(struct bt_hogp *hogp, struct bt_hogp_rep_info *rep, uint8_t err);
+
+static int send_hid_out_report(struct bt_hogp *hogp, const uint8_t *data, size_t size)
+{
+	uint8_t report_id = data[0];
+	struct bt_hogp_rep_info *out_rep = bt_hogp_rep_find(hogp,
+							    BT_HIDS_REPORT_TYPE_OUTPUT,
+							    report_id);
+
+	if (!out_rep) {
+		/* HID output report with given ID is not supported by the peripheral. */
+		return 0;
+	}
+
+	int err = bt_hogp_rep_write_wo_rsp(hogp, out_rep,
+					   &data[1], size - 1,
+					   hogp_out_rep_write_cb);
+
+	if (err == -ENOTCONN) {
+		LOG_INF("Cannot forward, peer disconnected");
+		err = 0;
+	}
+
+	return err;
+}
+
+static void hogp_out_rep_write_cb(struct bt_hogp *hogp, struct bt_hogp_rep_info *rep, uint8_t err)
+{
+	/* Make sure that the handler will not preempt system workqueue
+	 * and it will not be preempted by system workqueue.
+	 */
+	__ASSERT_NO_MSG(!k_is_in_isr());
+	__ASSERT_NO_MSG(!k_is_preempt_thread());
+
+	if (err) {
+		LOG_WRN("Failed to forward HID out report (err %" PRIu8 ")", err);
+	}
+
+	struct hids_peripheral *per = CONTAINER_OF(hogp, struct hids_peripheral, hogp);
+
+	/* Send update if it was queued. */
+	uint8_t sent_report_id = bt_hogp_rep_id(rep);
+	sys_snode_t *t;
```

Comment:
Please name the variables more significantly. Especially  `i`,` t`

Reply:
done

Reply:
It's much easier to read now. Thanks